### PR TITLE
Fixes #26502 - queries based on sync_plan changes in katello

### PR DESCRIPTION
**Pull request by kgaikwad**
In Katello, behavior of enabled field [changed](https://github.com/Katello/katello/commit/71f4d4a8871dc0448db674604f074a16b539804f).
It is not using `enabled` column from sync_plans table. 

I thought to use "hammer sync-plan list" command but `--organization-id` required there and which doesn't accept multiple values. so modified `sync_plans` feature to use SQL queries based on changes in Katello.

### CODE REVIEW

**Comment by jameerpathan111** [comment]
@kgaikwad thanks for quickly fixing this issue. :+1: 
I will test it soon and comment on PR with my findings.

**Comment by jameerpathan111** [comment]
@kgaikwad 
I have tested new changes on various satellite versions and it works as expected.
I didn't face any issue while testing this.
It LGTM.
```

# ./bin/foreman-maintain advanced procedure run sync-plans-disable
Running ForemanMaintain::Scenario
================================================================================
disable active sync plans: 
\ Total 4 sync plans are now disabled.                                [OK]      
--------------------------------------------------------------------------------
# ./bin/foreman-maintain advanced procedure run sync-plans-enable
Running ForemanMaintain::Scenario
================================================================================
re-enable sync plans: 
| Total 4 sync plans are now enabled.                                 [OK]      
--------------------------------------------------------------------------------
```

File: definitions/features/sync_plans.rb [hunk 1]
```diff
@@ -3,33 +3,49 @@ class Features::SyncPlans < ForemanMaintain::Feature
...
+        SQL
+      ).first['count'].to_i > 0
   end
 
   def ids_by_status(enabled = true)
-    enabled = enabled ? 't' : 'f'
-    feature(:foreman_database).query(
-      <<-SQL
-        SELECT id FROM katello_sync_plans WHERE enabled ='#{enabled}'
+    if required_new_implementation
```
Comment (mbacovsky) [inline definitions/features/sync_plans.rb]: @kgaikwad `ids_by_status` and `verify_existing_ids_by_status` are almost identical inside, could you please refator?

**Comment by kgaikwad** [comment]
Thank you @mbacovsky  for reviewing pull-request.
Refactored two methods into a single method in sync-plan feature.

@jameerpathan111,  it would be good to have one round of testing with new changes.

**Comment by mbacovsky** [comment]
@kgaikwad thanks for the refactoring it looks good to merge now. I'll merge it once @jameerpathan111 will give it a green light.

**Comment by jameerpathan111** [comment]
@kgaikwad @mbacovsky 

The new changes looks good and works as expected.
I have tried it on few different satellite versions and the issue seems to be fixed with this PR.
Here is what I tried:
- Create some sync plans(few of them disabled to begin with)
- Run `./bin/foreman-maintain advanced procedure run sync-plans-disable`
  - Active sync-plans got disabled.
- Run `./bin/foreman-maintain advanced procedure run sync-plans-enable`
  - Only the sync-plans which were disabled by `./bin/foreman-maintain advanced procedure run sync-plans-disable`, got enabled. It didn't updated sync-plans which were in disabled state to begin with.
```

# ./bin/foreman-maintain advanced procedure run sync-plans-disable
Running ForemanMaintain::Scenario
================================================================================
disable active sync plans: 
- Total 6 sync plans are now disabled.                                [OK]      
--------------------------------------------------------------------------------

# ./bin/foreman-maintain advanced procedure run sync-plans-enable
Running ForemanMaintain::Scenario
================================================================================
re-enable sync plans: 
- Total 6 sync plans are now enabled.                                 [OK]      
--------------------------------------------------------------------------------
```

**Comment by mbacovsky** [comment]
@jameerpathan111  thanks for the extensive testing!

**Pull request merged**